### DISCUSSION TITLE
ignore vcpkg Qt port and use manual/online installation instead

### DISCUSSION THREAD

**Discussion by FelixBer** [microsoft/vcpkg#26991] [General]
With Qt builds, it's common for people to have installed it via the online installer.
It works perfectly fine to use the online installer Qt in parallel with vcpkg libraries ... except when there is a rival Qt installed in vcpkg.
Then vcpk always prefers the vcpkg version over my manual installation, even when I try to force it by prepending the path to the manual installation to CMAKE_PREFIX_PATH or setting an explicit PATHS argument to find_package:

```
list(PREPEND CMAKE_PREFIX_PATH "C:/qt/6.3.0/msvc2019_64")
find_package(QT NAMES Qt6 Qt5 COMPONENTS Widgets REQUIRED PATHS "C:/qt/6.3.0/msvc2019_64")
find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Widgets REQUIRED PATHS "C:/qt/6.3.0/msvc2019_64")
message("Linking against Qt Version: "  ${Qt${QT_VERSION_MAJOR}_CONFIG})
# output: Linking against Qt Version: C:/vcpkg/installed/x64-windows/share/Qt6/Qt6Config.cmake
```

How can I force vcpkg to ignore its internal package and use my online installation instead?

**Comment by Neumann-A** [comment]
> How can I force vcpkg to ignore its internal package and use my online installation instead?

Simply use empty overlays for the Qt ports.

or use a customized triplet with per port customization and set `CURRENT_PORT_DIR` to a directory containing an empty portfile. (but that is just a quick hack and shouldn't be used ;) )

**Reply by FelixBer** [reply]
I considered that, but doesnt that require a re-installation of the package and then permanently overrides it? Meaning I would have to keep a separate vcpkg checkout for projects using the online installer - which is what I am doing right now.

https://vcpkg.readthedocs.io/en/latest/specifications/ports-overlay/ 
Didnt find a solution there, am I missing something?

**Reply by Neumann-A** [reply]
> Meaning I would have to keep a separate vcpkg checkout for projects using the online installer 

That is the whole point of vcpkg. You shouldn't mix and match different binaries. Also you don't need two different checkouts you just need two different installed trees (which would automatically exist if you use manifest mode).

**Reply by FelixBer** [reply]
I ensure binary compatibility. It's true that manifest mode would solve this problem, and I will look into switching to that. But for now we are stuck with "classical mode". Is there really no other way to mask a specific port in pure CMake? Perhaps by providing a custom FindQtX.cmake?

**Comment by dg0yt** [comment]
If you want to generally use particular alternative ports in your projects, you should look into registries, not overlay ports. A `vcpkg-configuration.json` in your project dir will specify where to find the registries and how to select ports.
[Creating Registries](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/registries.md) 
[Using Registries](https://github.com/microsoft/vcpkg/blob/master/docs/users/registries.md)

Note that Qt comes with bundled third-party libraries. You may get into a versioning and ODR issues if vcpkg and Qt provide different builds of libraries. There will also be a different pace on security updates. So be careful with mixing.
